### PR TITLE
Blog app - add forms

### PR DESCRIPTION
**Pull request by blase147**
In this project I;

- Created a method called current_user in my ApplicationController to make current user data available to all controllers.
    - It returns the first user from the database.
- Created forms to perform the following functions:
    - Create a Post on behalf of the current_user 
    - Create a comment on behalf of the current_user 
- Allowed Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = Post.find(params[:post_id])
+
+    respond_to do |_format|
+      if @comment.save
+
+        redirect_to user_post_path(user_id: params[:user_id], id: params[:post_id])
```
Comment (mckent05) [inline app/controllers/comments_controller.rb]: - Good job done so far, but this does not redirect to the user_post_path` when a new comment is created, please fix this 🙏

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -8,4 +8,29 @@ def show
...
+
+  def create
+    puts 'Current User'
+
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    respond_to do |_format|
+      if @post.save
+        redirect_to user_posts_path(current_user)
```
Comment (mckent05) [inline app/controllers/posts_controller.rb]: - According to the project requirements, when a new post is created it should redirect to the `user_post_path`, this is not so in your project, please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @blase147, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -1,25 +1,42 @@
 class CommentsController < ApplicationController
+  load_and_authorize_resource
```
Comment (gemmen29) [inline app/controllers/comments_controller.rb]: - [ ] Kindly remove the `load_and_authorize_resource` as it causes an error when I try to create a new comment.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @blase147, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Highlights💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi 🙋 @blase147   , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights </img>


- PR is professional ✔️ 
- Routing works as expected ✔️ 
- Linters checks passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**